apps/client/src/pages/QuestionsPage.js

```diff
@@ -46,34 +50,78 @@ const QuestionsPage = ({ topic, checker }) => {
     });
 
   const onQuestionNext = (value) => {
-    if (question.options) {
+    // Provide the user answers to the `sttr-checker`
+    if (question.options && value) {
       question.setAnswer(value);
-    } else {
+    }
+    if (!question.options && value) {
       const responseObj = booleanOptions.find((o) => o.formValue === value);
       question.setAnswer(responseObj.value);
     }
 
-    if (needContactPermits()) {
+    // Store all answers in the session context
+    sessionContext.setSessionData({
+      answers: checker.getQuestionAnswers(),
+    });
+
+    // Load next question
+    const next = checker.next();
+
+    // Go directly to the Conclusion Page, without passing the Results Page
+    // Only if the `sttr-checker` is the final question
+    if (needContactPermits() && !next) {
+      // Undo the next() with previous(), because we were already at the final question
+      checker.previous();
+
+      // Change the URL to the Conclusion Page
       history.push(geturl(routes.conclusion, { slug }));
     } else {
-      const next = checker.next();
+      // Load the next question or go to the Result Page
+      if (next) {
+        // Store the new questionIndex in the session
+        sessionContext.setSessionData({
+          questionIndex: sessionContext.questionIndex + 1,
+        });
 
-      if (!next) {
-        // Go to Result page
-        history.push(geturl(routes.results, { slug }));
-      } else {
         // Go to Next question
         setQuestion(next);
+
+        // Change the URL to the new question
+        history.push(
+          geturl(routes.questions, {
+            slug: topic.slug,
+            question: getslug(next.text),
+          })
+        );
+      } else {
+        // Go to Result page if there is no new quesion
+        history.push(geturl(routes.results, { slug }));
       }
     }
   };
 
   const onQuestionPrev = () => {
+    // Load the previous question or go to the Location Page
     if (checker?.stack?.length > 1) {
```

Comment:
Checker and checker.stack are guaranteed to be here. Please remove the ?'s.

Reply:
solved